compiler/src/dmd/dmangle.d

```diff
@@ -33,30 +34,37 @@ extern (C++) const(char)* mangleExact(FuncDeclaration fd)
 
 extern (C++) void mangleToBuffer(Type t, OutBuffer* buf)
 {
+    //printf("mangleToBuffer t()\n");
     if (t.deco)
         buf.writestring(t.deco);
     else
     {
-        scope Mangler v = new Mangler(buf, t);
-        v.visitWithMask(t, 0);
+        auto backref = Backref(t);
```

Comment:
Why is this kept outisde of the `mangleType` function? It doesn't really make any sense to put this burden on the caller. Just have `mangleType` receive a type as the class constructor did and instantiate the backref there.